x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportPutLifecycleAction.java

```diff
@@ -61,12 +64,17 @@ protected Response newResponse(boolean acknowledged) {
                     public ClusterState execute(ClusterState currentState) throws Exception {
                         ClusterState.Builder newState = ClusterState.builder(currentState);
                         IndexLifecycleMetadata currentMetadata = currentState.metaData().custom(IndexLifecycleMetadata.TYPE);
-                        if (currentMetadata.getPolicies().containsKey(request.getPolicy().getName())) {
+                        if (currentMetadata.getPolicyMetadatas().containsKey(request.getPolicy().getName())) {
                             throw new ResourceAlreadyExistsException("Lifecycle policy already exists: {}",
                                     request.getPolicy().getName());
                         }
-                        SortedMap<String, LifecyclePolicy> newPolicies = new TreeMap<>(currentMetadata.getPolicies());
-                        newPolicies.put(request.getPolicy().getName(), request.getPolicy());
+                        SortedMap<String, LifecyclePolicyMetadata> newPolicies = new TreeMap<>(currentMetadata.getPolicyMetadatas());
+
+                        Map<String, String> filteredHeaders = threadPool.getThreadContext().getHeaders().entrySet().stream()
+                                .filter(e -> ClientHelper.SECURITY_HEADER_FILTERS.contains(e.getKey()))
+                                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+                        LifecyclePolicyMetadata lifecyclePolicyMetadata = new LifecyclePolicyMetadata(request.getPolicy(), filteredHeaders);
```

Comment:
should we validate that only SECURITY_HEADER_FILTERS are passed to LifecyclePolicyMetadata? I'd imagine this is unnecessary once we would have qa tests with security enabled